### PR TITLE
1.15 : Repairing javadocs

### PR DESCRIPTION
**Pull request by Cyborgmas**
Doing my best at fixing old  javadocs. Did not just fix links, tried to make sure context was clear too.

To know the erroring javadocs I run the gradle task. Meaning that I am not fixing javadocs with old references that were not {@ link }  (and others).

### CODE REVIEW

**Comment by Cyborgmas** [comment]
Listing everything I did not do (that errors/warns when running javadoc)

- Add/complete @ params / @ returns / @ throws for javadocs that did not have it already (or empty ones). May have done 2-3, when it was clear what it was and provided good context.

- Repair the javadocs for : 
       - EntityEvent.CanUpdate
       - 
       - 
       - 
       -

**Comment by Cyborgmas** [comment]
Listing everything that errored/warned on javadoc task that I did not fix: 

- Incomplete/missing @ params / @ returns / @ throws 

- The following events, because they are never fired by forge
         - EntityEvent.CanUpdate
         - GetCollisionBoxesEvent
         - PlayerEvent.NameFormat 
         - FMLFingerprintViolationEvent
         - GuiScreenEvent.ActionPerformedEvent

- The BlockWrapper class, as I am very uncertain about what it does and what it was trying to 
  reference at the time and what it should reference now.

- In DeferredWorkQueue, not sure what to replace IThreadListener with.

- 2 Methods in IForgeBlock and IForgeBlockState that were marked with "TODO re-evaluate"
        - The getRayTraceResult method in IForgeBlock since it is unused and purpose unclear

- PlayerEntity#setSpawnPoint : The error is because the parameters are obfuscated. Do not want to 
  generate a patch to just obfuscate the @ param names

- @ apiNotes. the javadoc task did not recognize them , but they are valid so left them there

File: src/main/java/net/minecraftforge/event/world/WorldEvent.java [hunk 1]
```diff
@@ -81,10 +90,10 @@ public IWorld getWorld()
     /**
      * WorldEvent.Unload is fired when Minecraft unloads a world.<br>
      * This event is fired when a world is unloaded in
-     * {@link Minecraft#loadWorld(WorldClient, String)},
+     * {@link Minecraft#loadWorld(ClientWorld)},
      * {@link MinecraftServer#stopServer()},
-     * {@link DimensionManager#unloadWorlds()},
-     * {@link ForgeInternalHandler#onDimensionUnload(Unload)}. <br>
+     * {@link net.minecraftforge.common.DimensionManager#unloadWorlds},
+     * {@link net.minecraftforge.common.ForgeInternalHandler#onDimensionUnload}. <br>
```
Comment (Bricktricker) [inline src/main/java/net/minecraftforge/event/world/WorldEvent.java]: no longer called here, but called at [net.minecraft.client.Minecraft#unloadWorld](https://github.com/MinecraftForge/MinecraftForge/blob/d5cb0935e9758e1ae62dadf4535a12a5fda8e9f1/patches/minecraft/net/minecraft/client/Minecraft.java.patch#L333)

File: src/main/java/net/minecraftforge/event/entity/player/PlayerDestroyItemEvent.java [hunk 2]
```diff
@@ -34,24 +32,21 @@
...
+ * {@link PlayerEntity#attackTargetEntityWithCurrentItem(Entity)},
+ * {@link PlayerInteractionManager#tryHarvestBlock(BlockPos)}
  * {@link ForgeHooks#getContainerItem(ItemStack)},
- * {@link PlayerInteractionManager#processRightClick(PlayerEntity, World, ItemStack, Hand)},
- * {@link PlayerInteractionManager#processRightClickBlock(PlayerEntity, World, ItemStack, Hand, BlockPos, Direction, float, float, float)}
- * and {@link PlayerInteractionManager#tryHarvestBlock(BlockPos)}.<br>
+ * and {@link ForgeHooks#getContainerItem(ItemStack)}.<br>
  * <br>
  * {@link #original} contains the original ItemStack before the item was destroyed. <br>
  * (@link #hand) contains the hand that the current item was held in.<br>
```
Comment (Bricktricker) [inline src/main/java/net/minecraftforge/event/entity/player/PlayerDestroyItemEvent.java]: wrong brackets

File: src/main/java/net/minecraftforge/event/entity/living/LivingAttackEvent.java [hunk 3]
```diff
@@ -28,10 +28,10 @@
 /**
  * LivingAttackEvent is fired when a living Entity is attacked. <br>
  * This event is fired whenever an Entity is attacked in
- * {@link EntityLivingBase#attackEntityFrom(DamageSource, float)} and
- * {@link EntityPlayer#attackEntityFrom(DamageSource, float)}. <br>
+ * {@link LivingEntity#attackEntityFrom(DamageSource, float)} and
+ * {@link net.minecraft.entity.player.PlayerEntity#attackEntityFrom(DamageSource, float)}. <br>
```
Comment (Bricktricker) [inline src/main/java/net/minecraftforge/event/entity/living/LivingAttackEvent.java:32]: Also fired at
[RemoteClientPlayerEntity#attackEntityFrom(DamageSource, float)](https://github.com/MinecraftForge/MinecraftForge/blob/d5cb0935e9758e1ae62dadf4535a12a5fda8e9f1/patches/minecraft/net/minecraft/client/entity/player/RemoteClientPlayerEntity.java.patch#L7) and
[ClientPlayerEntity#attackEntityFrom(DamageSource, float)](https://github.com/MinecraftForge/MinecraftForge/blob/d5cb0935e9758e1ae62dadf4535a12a5fda8e9f1/patches/minecraft/net/minecraft/client/entity/player/ClientPlayerEntity.java.patch#L7)

File: src/main/java/net/minecraftforge/event/entity/living/LivingDropsEvent.java [hunk 4]
```diff
@@ -32,9 +31,9 @@
 /**
  * LivingDropsEvent is fired when an Entity's death causes dropped items to appear.<br>
  * This event is fired whenever an Entity dies and drops items in
- * {@link EntityLivingBase#onDeath(DamageSource)}.<br>
+ * {@link LivingEntity#onDeath(DamageSource)}.<br>
```
Comment (Bricktricker) [inline src/main/java/net/minecraftforge/event/entity/living/LivingDropsEvent.java]: No longer called here, now called at [LivingEntity#spawnDrops(DamageSource)](https://github.com/MinecraftForge/MinecraftForge/blob/d5cb0935e9758e1ae62dadf4535a12a5fda8e9f1/patches/minecraft/net/minecraft/entity/LivingEntity.java.patch#L187)

File: src/main/java/net/minecraftforge/event/entity/living/LivingEvent.java [hunk 5]
```diff
@@ -69,10 +69,10 @@ public LivingEntity getEntityLiving()
     /**
      * LivingJumpEvent is fired when an Entity jumps.<br>
      * This event is fired whenever an Entity jumps in
-     * {@link EntityLivingBase#jump()}, {@link EntityMagmaCube#jump()},
-     * and {@link EntityHorse#jump()}.<br>
+     * {@link LivingEntity#jump()}, {@link net.minecraft.entity.monster.MagmaCubeEntity#jump()},
+     * and {@link net.minecraft.entity.passive.horse.HorseEntity#jump()}.<br>
```
Comment (Bricktricker) [inline src/main/java/net/minecraftforge/event/entity/living/LivingEvent.java]: No longer fired at net.minecraft.entity.passive.horse.HorseEntity#jump()

File: src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java [hunk 6]
```diff
@@ -93,9 +93,9 @@ public HarvestCheck(PlayerEntity player, BlockState state, boolean success)
     /**
      * BreakSpeed is fired when a player attempts to harvest a block.<br>
      * This event is fired whenever a player attempts to harvest a block in
-     * {@link EntityPlayer#canHarvestBlock(IBlockState)}.<br>
+     * {@link PlayerEntity#canHarvestBlock(BlockState)}.<br>
```
Comment (Bricktricker) [inline src/main/java/net/minecraftforge/event/entity/player/PlayerEvent.java]: No longer called here, now called at [PlayerEntity#getDigSpeed(BlockState, BlockPos)](https://github.com/MinecraftForge/MinecraftForge/blob/d5cb0935e9758e1ae62dadf4535a12a5fda8e9f1/patches/minecraft/net/minecraft/entity/player/PlayerEntity.java.patch#L112)

File: src/main/java/net/minecraftforge/event/world/ChunkDataEvent.java [hunk 7]
```diff
@@ -89,7 +93,7 @@ public Load(IChunk chunk, CompoundNBT data, ChunkStatus.Type status)
     /**
      * ChunkDataEvent.Save is fired when vanilla Minecraft attempts to save Chunk data.<br>
      * This event is fired during chunk saving in
-     * {@link AnvilChunkLoader#saveChunk(World, Chunk)}. <br>
+     * {@link net.minecraft.world.server.ChunkManager#func_219229_a(IChunk)} )}. <br>
```
Comment (Bricktricker) [inline src/main/java/net/minecraftforge/event/world/ChunkDataEvent.java]: double `)}`

File: src/main/java/net/minecraftforge/event/world/ChunkEvent.java [hunk 8]
```diff
@@ -56,8 +59,8 @@ public IChunk getChunk()
     /**
      * ChunkEvent.Load is fired when vanilla Minecraft attempts to load a Chunk into the world.<br>
      * This event is fired during chunk loading in <br>
-     * {@link ChunkProviderClient#loadChunk(int, int)}, <br>
-     * Chunk.onChunkLoad(). <br>
+     * {@link net.minecraft.client.multiplayer.ClientChunkProvider#loadChunk(int, int, BiomeContainer, PacketBuffer, CompoundNBT, int)}, <br>
+     * {@link net.minecraft.world.server.ChunkManager#func_219229_a(IChunk)}<br>
```
Comment (Bricktricker) [inline src/main/java/net/minecraftforge/event/world/ChunkEvent.java]: thats `saveChunk`, `func_219200_b` should be the right one

File: src/main/java/net/minecraftforge/event/entity/living/LivingAttackEvent.java (hunk 3 above)
Comment (Cyborgmas) [reply src/main/java/net/minecraftforge/event/entity/living/LivingAttackEvent.java:32]: Well it is the same function in the end.
Just overriden

**Pull request closed without merging**